common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java

```diff
@@ -140,4 +165,39 @@ private String getCurrentActiveBrokerPackageName(@NonNull final Context context)
 
         return null;
     }
+
+    /**
+     * Returns the inactive installed authenticator.
+     * - If CP is the broker, this will return authenticator.
+     * - If authenticator is the broker, this will return CP.
+     *
+     * @param context
+     * @return package name of the installed inactive authenticator, if there's any.
+     */
+    private String getInactiveBrokerPackageName(@NonNull final Context context) {
+        final String methodName = ":getInactiveBrokerPackageName";
+        String currentBroker = getCurrentActiveBrokerPackageName(context);
+
+        final String inactiveBrokerPackageName;
+        if (currentBroker.equalsIgnoreCase(AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME)) {
+            inactiveBrokerPackageName = AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME;
+        } else if (currentBroker.equalsIgnoreCase(AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME)) {
+            inactiveBrokerPackageName = AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME;
+        } else {
+            return null;
+        }
+
+        // Verify the signature to make sure that we're not binding to malicious apps.
+        final BrokerValidator validator = new BrokerValidator(context);
+        try {
+            if (validator.verifySignature(inactiveBrokerPackageName)) {
+                return inactiveBrokerPackageName;
+            }
+        } catch (Exception e) {
```

Comment:
>Exception [](start = 17, length = 9)

I do not like this type of exception handling. I would like us to handle specific exceptions that are expected and then fallback on the 'Exception'.

Reply:
This one is closed. Please review https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/547 instead. Thanks!